src/main/java/com/philips/research/bombase/core/nuget/domain/NugetClient.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020-2021, Koninklijke Philips N.V., https://www.philips.com
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.philips.research.bombase.core.nuget.domain;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.github.packageurl.PackageURL;
+import com.philips.research.bombase.core.meta.PackageMetadata;
+import com.philips.research.bombase.core.nuget.NugetException;
+import org.springframework.stereotype.Component;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Locale;
+import java.util.Optional;
+
+@Component
+public class NugetClient {
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.NON_PRIVATE)
+            .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE);
+
+    private final NugetAPI rest;
+
+    NugetClient() {
+        this(URI.create("https://api.nuget.org/v3/"));
+    }
+
+    NugetClient(URI uri) {
+        final var retrofit = new Retrofit.Builder()
+                .baseUrl(uri.toASCIIString())
+                .addConverterFactory(JacksonConverterFactory.create(MAPPER))
+                .build();
+        rest = retrofit.create(NugetAPI.class);
+    }
+
+    Optional<PackageMetadata> getPackageMetadata(PackageURL purl) {
+        Optional<NugetAPI.CatalogResponseJson> optionalCatalogResponseJson = query(rest.getCatalogEntry(purl.getName().toLowerCase(Locale.ROOT), purl.getVersion()));
```

Comment:
A rule I try to follow for variable names is "short scope, short name". 

- I am not sure if the "optional" prefix adds value to explain what the variable holds, because the IDE nowadays helps in telling the typist about the available methods.)
- I am wondering if the "Json" postfix adds value, given that nothing in the method appears to be related to JSON processing.

(In this case I would personally opt for naming it simply `response`.)

Reply:
Thanks for the feedback, I will keep this in mind for the future. Often when I work in draft PR's, I tend to avoid thinking too much about variable naming as I will do this during refactoring.